File: app/src/main/java/com/example/projetkotlin/Sauvegarde.java
package com.example.projetkotlin;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class Sauvegarde {
    private static final String LIST_KEY="list_key";
    public static void writeListinPref(Context context, List<ToDo> list) {
        Gson gson = new Gson();
        String jsonString = gson.toJson(list);
        SharedPreferences pref = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor editor = pref.edit();
        editor.putString(LIST_KEY, jsonString);
        editor.apply();
    }
    public static List<ToDo> readListFromPref(Context context){
        SharedPreferences pref = PreferenceManager.getDefaultSharedPreferences(context);
        String jsonString = pref.getString(LIST_KEY,"");
        Gson gson = new Gson();
        Type type = new TypeToken<ArrayList<ToDo>>(){}.getType();
        List<ToDo> list = gson.fromJson(jsonString,type);
        return list;
    }
}

File: app/src/main/java/com/example/projetkotlin/Sauvegarde.kt
package com.example.projetkotlin

import android.content.Context
import com.google.gson.Gson
import android.preference.PreferenceManager
import com.google.gson.reflect.TypeToken
import kotlin.collections.ArrayList

object Sauvegarde {
    private const val LIST_KEY = "list_key"
    fun writeListinPref(context: Context?, list: ArrayList<ToDo>) {
        val gson = Gson()
        val jsonString = gson.toJson(list)
        val pref = PreferenceManager.getDefaultSharedPreferences(context)
        val editor = pref.edit()
        editor.putString(LIST_KEY, jsonString)
        editor.apply()
    }

    fun readListFromPref(context: Context?): ArrayList<ToDo> {
        val pref = PreferenceManager.getDefaultSharedPreferences(context)
        val jsonString = pref.getString(LIST_KEY, "")
        val gson = Gson()
        val type = object : TypeToken<ArrayList<ToDo?>?>() {}.type
        val list = gson.fromJson<ArrayList<ToDo>>(jsonString, type)
        val list2: ArrayList<ToDo> = ArrayList()
        return list ?: list2
    }


}